RR.App/Controllers/HRIS/EmployeeController.cs

```diff
@@ -68,181 +56,61 @@ public async Task<IActionResult> GetEmployeeById([FromQuery] int id)
     [HttpGet("by-email")]
     public async Task<IActionResult> GetEmployeeByEmail([FromQuery] string? email)
     {
-        try
-        {
-            var claimsIdentity = User.Identity as ClaimsIdentity;
-
-            var emailToUse = email ??
-                             claimsIdentity!.FindFirst(ClaimTypes.Email)!.Value;
-
-            var employee = await _employeeService.GetEmployee(emailToUse);
 
+            var employee = await _employeeService.GetEmployee(_identity.Email);
             return Ok(employee);
-        }
-        catch (Exception ex)
-        {
-            return NotFound(ex.Message);
-        }
+
     }
 
     [Authorize(Policy = "AllRolesPolicy")]
     [HttpPut]
     public async Task<IActionResult> UpdateEmployee([FromBody] EmployeeDto employee, [FromQuery] string userEmail)
     {
-        try
-        {
-            if (_identity.Role is "SuperAdmin" or "Admin" or "Talent" or "Journey")
-            {
-                var updatedEmployee = await _employeeService.UpdateEmployee(employee, _identity.Email);
-                return CreatedAtAction(nameof(UpdateEmployee), new { email = updatedEmployee.Email }, updatedEmployee);
-            }
-
-            if (employee.Id == _identity.EmployeeId)
-            {
-                var updatedEmployee = await _employeeService.UpdateEmployee(employee, _identity.Email);
-                return CreatedAtAction(nameof(UpdateEmployee), new { email = updatedEmployee.Email }, updatedEmployee);
-            }
-
-            return NotFound("User data being accessed does not match user making the request.");
-        }
-        catch (Exception ex)
-        {
-            if (ex.Message.Contains("Unauthorized action"))
-                return StatusCode(403, $"Forbidden: {ex.Message}");
-            return NotFound(ex.Message);
-        }
+
+            if (_identity.Role is not "SuperAdmin" or "Admin" or "Talent" or "Journey" && employee.Id != _identity.EmployeeId)
```

Comment:
```suggestion
            if (_identity.Role is "SuperAdmin" or "Admin" or "Talent" or "Journey" == false && employee.Id != _identity.EmployeeId)
```

If you use the statement `role is not "value" or "other"`. It will always only check `role is not "value"` and ignore the `or "other"` part. The alternative way is have a false check to compare it against